src/Groups/pcgroup.jl

```diff
@@ -0,0 +1,345 @@
+  i < j || throw(ArgumentError("only for i < j, but i = $i, j = $j"))
+  c.conjugates[i,j] = copy(rhs)
+
+  if isdefined(c, :X)
+    if GAP.Globals.IsSingleCollectorRep(c.X)
+      G = c.F
+      GAP.Globals.SetConjugate(c.X, j, i, G(rhs).X)
+    elseif GAP.Globals.IsFromTheLeftCollectorRep(c.X)
+      GAP.Globals.SetConjugate(c.X, j, i, _GAP_generator_exponent_vector(rhs))
+    else
+      error("unknown GAP collector")
+    end
+  end
+end
+
+# Create a collector independent of `c`.
+function Base.deepcopy_internal(c::GAP_Collector{T}, dict::IdDict) where T <: IntegerUnion
+  cc = GAP_Collector{T}(c.ngens,
+         deepcopy_internal(c.relorders, dict::IdDict),
+         deepcopy_internal(c.powers, dict::IdDict), 
+         deepcopy_internal(c.conjugates, dict::IdDict))
+
+  if isdefined(c, :X)
+    if GAP.Globals.IsSingleCollectorRep(c.X)
+      # For this type of collectors, `GAP.Globals.ShallowCopy`
+      # returns an independent copy.
+      cc.X = GAP.Globals.ShallowCopy(c.X)
+    elseif GAP.Globals.IsFromTheLeftCollectorRep(c.X)
+      # Currently there is no `GAP.Globals.ShallowCopy` method for `c.X`.
+      # Create the GAP object anew.
+      cc.X = _GAP_collector_from_the_left(c)
+    else
+      error("unknown GAP collector")
+    end
+  end
+
+  return cc
+end
+
+# Create a new GAP collector using `GAP.Globals.SingleCollector`.
+function _GAP_single_collector(c::GAP_Collector)
+  G = free_group(c.ngens; eltype = :syllable)
+
+  cGAP = GAP.Globals.SingleCollector(G.X, GapObj(c.relorders, recursive = true))::GapObj
+  for i in 1:c.ngens
+    GAP.Globals.SetPower(cGAP, i, G(c.powers[i]).X)
+  end
+  for j in 1:c.ngens
+    for i in 1:(j-1)
+      if isassigned(c.conjugates, i, j)
+        GAP.Globals.SetConjugate(cGAP, j, i, G(c.conjugates[i, j]).X)
+      else
+        GAP.Globals.SetConjugate(cGAP, j, i, G([j => 1]).X)
+      end
+    end
+  end
+  GAP.Globals.UpdatePolycyclicCollector(cGAP)
+
+  return cGAP, G
+end
+
+
+# Create a new GAP collector using `GAP.Globals.FromTheLeftCollector`
+# (from GAP's Polycyclic package).
+function _GAP_collector_from_the_left(c::GAP_Collector)
+  cGAP = GAP.Globals.FromTheLeftCollector(c.ngens)
+  for i in 1:c.ngens
+    if c.relorders[i] != 0
+      # We are not allowed to set relative orders to zero.
+      GAP.Globals.SetRelativeOrder(cGAP, i, GAP.Obj(c.relorders[i]))
+    end
+  end
+  for i in 1:c.ngens
+    if c.relorders[i] != 0
+      GAP.Globals.SetPower(cGAP, i, _GAP_generator_exponent_vector(c.powers[i]))
+    end
+  end
+  for j in 1:c.ngens
+    for i in 1:(j-1)
+      if isassigned(c.conjugates, i, j)
+        GAP.Globals.SetConjugate(cGAP, j, i, _GAP_generator_exponent_vector(c.conjugates[i, j]))
+      else
+        GAP.Globals.SetConjugate(cGAP, j, i, GapObj([j, 1]))
+      end
+    end
+  end
+  GAP.Globals.UpdatePolycyclicCollector(cGAP)
+  
+  return cGAP
+end
+
+
+# Create the collector on the GAP side on demand
+function Base.getproperty(c::GAP_Collector, sym::Symbol)
```

Comment:
We use `getproperty` like this for some of the other types mostly for legacy reasons, IMHO (namely because we did not provide a good way to access the underlying GAP object).

I think it would be preferable to change code which uses `collector.X` to get the underlying GAP object to use a helper method instead. The canonical idea would be to defined `GAPObj(collector)` for this role (i.e., what the code here does: compute `collector.X` if missing and then return it). Though we could also discuss other names, of course (`underlying_gap_object`?)